File: 1. 단계별로_풀어보기/4. 1차원 배열/5. 10810.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N, M = map(int, input().split())

basketArr = [0 for i in range(N)]

for idx in range(M):
  i, j, k = map(int, input().split())

  basketArr[i:j + 1] = [k] * (j - i + 1)

output(f"{" ".join(str(basketArr[i]) for i in range(N))}")

File: 1. 단계별로_풀어보기/1~9/4. 1차원 배열/5. 10810.js
const fs = require('fs')
const inputs = fs.readFileSync('/dev/stdin').toString().split('\n')

const [N, M] = inputs[0].split(' ').map(Number)

const basketArr = Array(N).fill(0)

for (let idx = 0; idx < M; idx++) {
  const [i, j, k] = inputs[idx + 1].split(' ').map(Number)

  for (let bIdx = i - 1; bIdx < j; bIdx++) {
    basketArr[bIdx] = k
  }
}

process.stdout.write(`${basketArr.join(' ')}`)
